accounts/README.md

```diff
@@ -1,10 +1,121 @@
 # Accounts
 
-The Safe{Core} Protocol is designed to be **account agnostic**. This initial alpha version sets a focus on the 1.x versions of Safe Smart Accounts to expedite the development process and gather feedback. These learnings are the foundation upon which the protocol is opened up to other account implementations.
+The Safe{Core} Protocol is designed to be **account agnostic**. This initial alpha version focuses on the 1.x versions of Safe Smart Accounts to expedite the development process and gather feedback. These learnings are the foundation upon which the protocol is opened to other account implementations.
 
-## Safe 1.4.0
+## High-level overview
 
-[Source Code](https://github.com/safe-global/safe-contracts/tree/v1.4.0)
+The [Safe{Core} Protocol Manager](https://github.com/safe-global/safe-core-protocol-specs/blob/2bffd759dd12be5583594f302d97c35e0ab9fcf5/manager/README.md) contract is the main entity interacting with accounts and vice versa. The Manager expects accounts to implement a specific interface, described in this document.
+
+## API Specification
+
+### General requirements
+
+The Account MUST append the 20 bytes of the `msg.sender` address to the calldata to each Account configuration-related method (setting a hook,
+adding a plugin, etc., for the full list, refer to [Manager's](../manager/README.md) specification). This is required because
+to support some functionality, such as function handlers, an Account may be required to execute a CALL
+operation that potentially bypasses the authorization to the Manager contract. This is required to allow the Manager to identify the transaction's sender. It is assumed
+that the only way to have the `msg.sender` equal to the Account address in the call frame is to execute a call through the Account's built-in authorization scheme.
```

Comment:
Is adding `msg.sender` completely covered by the reference `fallback` implementation?

The reason I ask is that I would propose not having this requirement:
1. If the `Account` implementation is expected to internally make configuration calls like adding a plugin, then I would change the `Manager.addPlugin` implementation to take an explicit caller parameter instead of having this "rule"
2. If the `Account` calls configuration function through its fallback handler, then this is already specified through the reference `fallback` implementation right?

Sorry for being nit-picky but this paragraph "felt a bit off" to me and it's the spec 😅 which needs to be 👌.

Reply:
> That means an arbitrary sender can call into the account and execute an arbitrary CALL to the Manager. Because of this, we had to add this requirement for the Manager to authenticate the caller.

I don't think this is a form of authentication in the Manager side though - a malicious account implementation or caller can just as easily add a different address at the end of the Manager call.

My point was more about questioning "why is this requirement needed". If the configuration change methods are always called via the `fallback` mechanism, then this is a non-issue correct (as in, this requirement doesn't need to be mentioned as it is captured by the `fallback` mechanism specification)?

Is my understanding correct that if the `Account` has code where it internally makes configuration changes, then it needs to either:
- `IManager(this).configurationMethod()` (to ensure that it goes over the fallback mechanism, at the cost of an additional `CALL`, which I think _should_ be fine in the name of simplicity)
- Manually add the `msg.sender` at the end of the call data, in which case it likely makes sense to expose methods for configuration in the form of a library:

```solidity
library ManagerLib {
    function addPluginInternal(IManager manager, ...) {
        bytes memory callData = abi.encodeCall(manager.addPluginInternal, (...));
        manager.call(callData);
    }
}

contract Account is IAccount {
    using ManagerLib for IManager;
    
    IManager manager;
    
    function doSomeConfiguration() {
        manager.addPluginInternal(...);
    }
}
```

And in this context, this requirement is meant to capture the requirement for the second option right?